SIGSEGV flake in TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown

From https://github.com/googleforgames/agones/pull/2979#issuecomment-1441100370:

```
generic-1.24: Step #1 - "e2e-feature-gates": --- FAIL: TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown (0.15s)
generic-1.24: Step #1 - "e2e-feature-gates": panic: runtime error: invalid memory address or nil pointer dereference [recovered]
generic-1.24: Step #1 - "e2e-feature-gates": 	panic: runtime error: invalid memory address or nil pointer dereference
generic-1.24: Step #1 - "e2e-feature-gates": [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x20f9f70]
generic-1.24: Step #1 - "e2e-feature-gates": 
generic-1.24: Step #1 - "e2e-feature-gates": goroutine 9 [running]:
generic-1.24: Step #1 - "e2e-feature-gates": testing.tRunner.func1.2({0x22a4200, 0x340b020})
generic-1.24: Step #1 - "e2e-feature-gates": 	/usr/local/go/src/testing/testing.go:1396 +0x372
generic-1.24: Step #1 - "e2e-feature-gates": testing.tRunner.func1()
generic-1.24: Step #1 - "e2e-feature-gates": 	/usr/local/go/src/testing/testing.go:1399 +0x5f0
generic-1.24: Step #1 - "e2e-feature-gates": panic({0x22a4200, 0x340b020})
generic-1.24: Step #1 - "e2e-feature-gates": 	/usr/local/go/src/runtime/panic.go:890 +0x262
generic-1.24: Step #1 - "e2e-feature-gates": agones.dev/agones/test/e2e/extensions.TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown(0xc0002f6d00)
generic-1.24: Step #1 - "e2e-feature-gates": 	/go/src/agones.dev/agones/test/e2e/extensions/high_availability_test.go:56 +0x490
generic-1.24: Step #1 - "e2e-feature-gates": testing.tRunner(0xc0002f6d00, 0x258d3f0)
generic-1.24: Step #1 - "e2e-feature-gates": 	/usr/local/go/src/testing/testing.go:1446 +0x217
generic-1.24: Step #1 - "e2e-feature-gates": created by testing.(*T).Run
generic-1.24: Step #1 - "e2e-feature-gates": 	/usr/local/go/src/testing/testing.go:1493 +0x75e
generic-1.24: Step #1 - "e2e-feature-gates": FAIL test/e2e/extensions.TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown (0.15s)
generic-1.24: Step #1 - "e2e-feature-gates": FAIL test/e2e/extensions
```

--- a/test/e2e/extensions/high_availability_test.go
+++ b/test/e2e/extensions/high_availability_test.go
@@ -22,6 +22,7 @@ import (
 	"agones.dev/agones/pkg/util/runtime"
 	e2eframework "agones.dev/agones/test/e2e/framework"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
@@ -41,18 +42,23 @@ func TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown(t *testing.T) {
 
 	list, err := getAgoneseExtensionsPods(ctx)
 	logger.Infof("Length of pod list is %v", len(list.Items))
-	assert.NoError(t, err, "Could not get list of Extension pods")
+	for i := range list.Items {
+		logger.Infof("Name of extensions pod %v: %v", i, list.Items[i].ObjectMeta.Name)
+		logger.Infof("Host IP %v", list.Items[i].Status.HostIP)
+		logger.Infof("Pod IPs %v", list.Items[i].Status.PodIPs)
+	}
+	require.NoError(t, err, "Could not get list of Extension pods")
 	assert.Greater(t, len(list.Items), 1, "Cluster has no Extensions pod or has only 1 extensions pod")
 
-	logger.Info("Removing one of the Extensions Pods")
-	deleteAgonesExtensionsPods(ctx, t)
+	logger.Infof("Removing one of the Extensions Pods: %v", list.Items[1].ObjectMeta.Name)
+	deleteAgonesExtensionsPod(ctx, t)
 
-	assert.NoError(t, waitForAgonesExtensionsRunning(ctx))
+	require.NoError(t, waitForAgonesExtensionsRunning(ctx))
 
-	logger.Info("Creating default game server")
 	gs := framework.DefaultGameServer(defaultNs)
+	logger.Infof("Creating game-server %s...", gs.Name)
 	readyGs, err := framework.CreateGameServerAndWaitUntilReady(t, defaultNs, gs)
-	assert.NoError(t, err, "Could not get a GameServer ready")
+	require.NoError(t, err, "Could not get a GameServer ready")
 	logger.WithField("gsKey", readyGs.ObjectMeta.Name).Info("GameServer Ready")
 
 	assert.NoError(t, framework.AgonesClient.AgonesV1().GameServers(defaultNs).Delete(ctx, readyGs.ObjectMeta.Name, metav1.DeleteOptions{})) // nolint: errcheck
@@ -60,14 +66,19 @@ func TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown(t *testing.T) {
 
 // deleteAgonesExtensionsPod deletes one of the extensions pod for the Agones extensions,
 // faking a extensions pod crash.
-func deleteAgonesExtensionsPods(ctx context.Context, t *testing.T) {
+func deleteAgonesExtensionsPod(ctx context.Context, t *testing.T) {
 	list, err := getAgoneseExtensionsPods(ctx)
 	assert.NoError(t, err)
 
 	policy := metav1.DeletePropagationBackground
-	err = framework.KubeClient.CoreV1().Pods("agones-system").Delete(ctx, list.Items[1].ObjectMeta.Name,
+	podToDelete := list.Items[1]
+	err = framework.KubeClient.CoreV1().Pods("agones-system").Delete(ctx, podToDelete.ObjectMeta.Name,
 		metav1.DeleteOptions{PropagationPolicy: &policy})
 	assert.NoError(t, err)
+	require.Eventually(t, func() bool {
+		_, err := framework.KubeClient.CoreV1().Pods("agones-system").Get(ctx, podToDelete.ObjectMeta.Name, metav1.GetOptions{})
+		return err != nil
+	}, 5*time.Minute, time.Second)
 }
 
 func waitForAgonesExtensionsRunning(ctx context.Context) error {
@@ -75,6 +86,10 @@ func waitForAgonesExtensionsRunning(ctx context.Context) error {
 		list, err := getAgoneseExtensionsPods(ctx)
 		if err != nil {
 			return true, err
+		}
+
+		if len(list.Items) != 2 {
+			return false, nil
 		}
 
 		for i := range list.Items {

